src/dvc_data/index/diff.py

```diff
@@ -255,35 +256,39 @@ def _detect_renames(changes: Iterable[Change]):
     def _get_key(change):
         return change.key
 
-    added[:] = sorted(added, key=_get_key)
-    deleted[:] = sorted(deleted, key=_get_key)
+    # Sort the lists to maintain the same order
+    # as older implementation.
```

Comment:
I don't think this comment is important. Let's remove it.

Reply:
What are you thinking about sorting in general? The only reason I included it, is to not alter the previous order. Is that important, or should we remove the sorting entirely?

Reply:
yes, order is important for `dvc diff` (either here or in dvc side).

Reply:
Just to make sure I understand: the order will be deterministic even without the sort, right? So the sort is only to have `dvc diff` give the same output as the previous version.

Reply:
Internally, it's hard to say if it's going to be deterministic or not. Let's say `ls()` (our internal index implementation) may end up giving different results in a different order. If it's deterministic by some chance, it's not something to be relied upon.

`diff` needs to display in a natural order (or a lexicographic order) that the user expects.

Reply:
Ok, thanks for the clarification:)